### PR TITLE
Avenmia/connect survey to db

### PR DESCRIPTION
**Pull request by avenmia**
This should connect the placeholder survey to the database as well as allow the user to submit their answers to the database

Here is a screenshot of the schema for how the questions, answers, and user answers are currently arranged.

image

The position column in the SurveyQuestion and SurveyAnswers allows for them to customize the order of the questions and answers. 

In order to populate the database, I seeded the tables as shown in the migration file that was added

The code is messy since the IDs are guids and I'm only storing IDs in the UserAnswers table. If there is a better way to organize or implement let me know.

### CODE REVIEW

File: src/components/survey/demographicssurvey.tsx [hunk 1]
```diff
@@ -2,40 +2,46 @@ import React, { useCallback, useMemo, useState } from "react";
 import SurveyQuestion from "./surveyquestion";
 import Link from "next/link";
 import { NextPageButtonLink } from "../../UI/NextPageButtonLink";
+import { api } from "../../utils/api";
 
 interface SurveyData {
+  questionId: string;
   question: string;
-  answers: string[]
+  answers: {answer:string, answerId: string}[]
```
Comment (tyliec) [inline src/components/survey/demographicssurvey.tsx:10]: Can we type this separately?

```javascript
type SurveyAnswer {
  answer: string,
  answerId: string
}
```

File: src/server/api/routers/survey.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import { z } from "zod";
+
+import { createTRPCRouter, publicProcedure } from "../trpc";
+
+export const surveyRouter = createTRPCRouter({
+  getSurveyData: publicProcedure
```
Comment (tyliec) [inline src/server/api/routers/survey.ts:6]: Can we split these into imported/exported functions? something like

```javascript
import { getSurveyData } from "../getSurveyData
...
export const surveyRouter = createTRPCRouter({
  getSurveyData,
  ...
});
```

File: src/server/api/routers/survey.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  addUserAnswer: publicProcedure
+    .input(z.object({ answerId: z.string(), questionId: z.string() }))  
+    .mutation(async ({input, ctx}) => {
+      if(!ctx.session){
+        console.log("Not authenticated")
+        return null;
+      }
+      const { id: userId } = ctx.session.user
+      const answerId = input.answerId;
+      const questionId = input.questionId;
```
Comment (tyliec) [inline src/server/api/routers/survey.ts:29]: This can be something like const { answerId, questionId } = input;

File: src/pages/polis.tsx [hunk 4]
```diff
@@ -9,9 +9,9 @@ const Polis: NextPage = () => {
     <div className="bg-[#3276AE] flex flex-col items-center h-screen">
       <h1 className="text-center text-2xl text-white mb-10">Please select the Pol.is survey you would wish to complete.</h1>
       {
-        surveys?.map(survey => {
+        surveys?.map((survey,index) => {
           return (
-            <NextPageButtonLink pageName="polissurvey" msg={survey} query={{surveyId: survey}} />
+            <NextPageButtonLink key={index} pageName="polissurvey" msg={survey} query={{surveyId: survey}} />
```
Comment (tyliec) [inline src/pages/polis.tsx:14]:
```suggestion
            <NextPageButtonLink key="survey-{index}" pageName="polissurvey" msg={survey} query={{surveyId: survey}} />
```

File: src/server/api/routers/survey.ts (hunk 2 above)
Comment (avenmia) [reply src/server/api/routers/survey.ts:6]: I'm not sure how to pull these out of the router

**Review by tyliec — APPROVED** [review]

**Pull request merged**